### DISCUSSION TITLE
Remote table reference but do the computation locally

### DISCUSSION THREAD

**Discussion by kellwu** [ClickHouse/ClickHouse#50633] [Q&A] [answered]
I want to be able to access the same databases on two servers for computation (let's say A and B).
Option1:
  - create the tables on A and store on s3 `create ... settings disk='s3'`
  - config the same s3 bucket path as disk on server B and somehow reference the table in s3?
Problem: I checked the table files on s3 and they are letters. Looks like only the clickhouse server A (the server that created those files) knows how to read them. **I don't know whether we can and how to do the second step**

Option2:
  - use remote table
Problem: the computation will only occur on the server with the data

Option3:
  - zero replication
Problem: just an experimental feature

Option4:
  - create tables on A
  -  setup <remote_server> on B and config A as the shard
  - create a distribution table on B to query data on A
I think this is similar to option2 but distribution table will handle the computation so I am able to use the CPU resources on B

I feel Option1 is the easiest to setup in my case, does anyone know if it is achievable?
Also, I am not sure if option4 has any downside that I haven't notice; Looks like distribution normally used in clusters and I am not sure if it is also ok to use in this case

**Comment by DanRoscigno** [comment]
Hi @kellwu I think that you want to use the [Distributed Table Engine](https://clickhouse.com/docs/en/engines/table-engines/special/distributed) .  I think that when you create the table and specify "ON CLUSTER clustername" and "ENGINE = Distributed" the table will be visible from all of the cluster members.

**Comment by DanRoscigno** [comment, marked as answer]
@kellwu have a look at this PR: https://github.com/ClickHouse/examples/pull/22
